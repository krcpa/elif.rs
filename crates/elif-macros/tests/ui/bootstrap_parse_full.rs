// Test that bootstrap macro parsing works with parameters
#[elif::bootstrap(addr = "0.0.0.0:8080")]
async fn main() -> Result<(), HttpError> {
    // This should parse correctly (auto-discovery with custom address)
}