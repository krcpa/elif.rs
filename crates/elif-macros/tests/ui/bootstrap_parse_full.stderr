error[E0433]: failed to resolve: use of unresolved module or unlinked crate `tokio`
 --> tests/ui/bootstrap_parse_full.rs:2:1
  |
2 | #[elif::bootstrap(addr = "0.0.0.0:8080")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `tokio`
  |
  = note: this error originates in the attribute macro `elif::bootstrap` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `elif_http`
 --> tests/ui/bootstrap_parse_full.rs:2:1
  |
2 | #[elif::bootstrap(addr = "0.0.0.0:8080")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `elif_http`
  |
  = help: if you wanted to use a crate named `elif_http`, use `cargo add elif_http` to add it to your `Cargo.toml`
  = note: this error originates in the attribute macro `elif::bootstrap` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this struct
  |
2 + use elif::prelude::AppBootstrapper;
  |

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `env_logger`
 --> tests/ui/bootstrap_parse_full.rs:2:1
  |
2 | #[elif::bootstrap(addr = "0.0.0.0:8080")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `env_logger`
  |
  = help: if you wanted to use a crate named `env_logger`, use `cargo add env_logger` to add it to your `Cargo.toml`
  = note: this error originates in the attribute macro `elif::bootstrap` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0752]: `main` function is not allowed to be `async`
 --> tests/ui/bootstrap_parse_full.rs:2:1
  |
2 | #[elif::bootstrap(addr = "0.0.0.0:8080")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`
  |
  = note: this error originates in the attribute macro `elif::bootstrap` (in Nightly builds, run with -Z macro-backtrace for more info)
